### PR TITLE
chore: GCP operation tracker refactor

### PR DESCRIPTION
**Pull request by erikwilson**
## Description

Remove use of actors for the GCP operation tracker. 

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
DET-9625

### CODE REVIEW

**Review by maxrussell — APPROVED** [review]
Couple of nits, but LGTM

File: master/internal/rm/agentrm/provisioner/gcp_operation_tracker.go [hunk 1]
```diff
@@ -26,171 +24,159 @@ type (
...
 
-	case trackOperationDone:
-		if msg.err != nil {
-			ctx.Log().WithError(msg.err).Error("")
+func (t *gcpBatchOperationTracker) startTracker(postProcess func([]*compute.Operation)) {
+	timeout := time.Duration(t.config.OperationTimeoutPeriod)
+	groupCtx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	g := new(errgroup.Group)
```
Comment (stoksc) [inline master/internal/rm/agentrm/provisioner/gcp_operation_tracker.go:121]: i think `var g errgroup.Group` is a way i see to use a zero value more frequently.

File: master/internal/rm/agentrm/provisioner/gcp_operation_tracker.go [hunk 2]
```diff
@@ -26,171 +24,159 @@ type (
...
+			done(t.op, resp, nil)
+			return nil
+		case resp.Status == "RUNNING" || resp.Status == "PENDING":
+			// Do nothing, keep tracking the operation.
+		default:
+			errOp, _ := json.Marshal(resp)
+			return errors.Errorf("unexpected message: %s", errOp)
+		}
+		// Slow down the polling rate.
+		time.Sleep(time.Second)
```
Comment (stoksc) [inline master/internal/rm/agentrm/provisioner/gcp_operation_tracker.go:86]: `time.Ticker` instead? so you check for context timeouts while sleeping.

**Review by stoksc — APPROVED** [review]
lgtm. two minor comments, other than that i agree we could remove these.

**Pull request merged**